Include/internal/pycore_signal.h

```diff
@@ -38,12 +37,16 @@ PyAPI_FUNC(void) _Py_RestoreSignals(void);
 #define INVALID_FD (-1)
 
 struct _signals_runtime_state {
-    volatile struct {
-        _Py_atomic_int tripped;
+    struct {
+        /* To access tripped / func with "relaxed" memory order
```

Comment:
https://github.com/python/cpython/pull/110480#discussion_r1349538339

@vstinner Enough or do you want to explain in other way?

Reply:
Hum, maybe move this comment at the top of trip_signal(), since it's the function which writes into these two variables.

Reply:
I'm mostly curious about setting tripped and then set is_tripped. How/why it's ok to use relaxed to set tripped, but why is_tripped requires a memory barrier.

I don't get why you mention "func" here, since get_handler() and set_handler() don't use relaxed but "sequentially consistent" order to get/set it.

Reply:
> I don't get why you mention "func" here, since get_handler() and set_handler() don't use relaxed but "sequentially consistent" order to get/set it.

Mistake

Reply:
> I'm mostly curious about setting tripped and then set is_tripped. How/why it's ok to use relaxed to set tripped, but why is_tripped requires a memory barrier.

Hmm change my mind, let's change the implementation of the usage.
It looks quite an implementation detail, so can make mistakes anytime.